examples/java8/src/test/java/org/apache/beam/examples/snippets/SnippetsTest.java

```diff
@@ -42,7 +45,29 @@
 public class SnippetsTest implements Serializable {
 
   @Rule
-  public transient TestPipeline p = TestPipeline.create();
+  public final transient TestPipeline p = TestPipeline.create();
+
+  @Test
+  public void testModelBigQueryIO() {
+    // We cannot test BigQueryIO functionality in unit tests, therefore we limit ourselves
+    // to making sure the pipeline containing BigQuery sources and sinks can be built.
+    boolean runLocally = false;
+    if (runLocally) {
+      String project = "my-project";
+      String dataset = "samples";  // this must already exist
+      String table = "modelBigQueryIO";  // this will be created if needed
+
+      BigQueryOptions options = PipelineOptionsFactory.create().as(BigQueryOptions.class);
+      options.setProject(project);
+      options.setTempLocation("gs://" + project + "/samples/temp/");
+      Pipeline p = Pipeline.create(options);
+      Snippets.modelBigQueryIO(p, project, dataset, table);
+      p.run();
+    } else {
+      Pipeline p = Pipeline.create();
+      Snippets.modelBigQueryIO(p);
```

Comment:
p.run() ?

Reply:
Running this will fail due to credentials and table permissions. Therefore we limit to making sure the transforms can be built and that every validation passes. See comment at line 52.